polaris-service/src/main/java/org/apache/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -1342,24 +1377,45 @@ public void doCommit(TableMetadata base, TableMetadata metadata) {
       TableLikeEntity entity =
           TableLikeEntity.of(resolvedEntities == null ? null : resolvedEntities.getRawLeafEntity());
       String existingLocation;
+      int maxMetadataCacheBytes =
+          callContext
+              .getPolarisCallContext()
+              .getConfigurationStore()
+              .getConfiguration(
+                  callContext.getPolarisCallContext(),
+                  catalogEntity,
+                  PolarisConfiguration.METADATA_CACHE_MAX_BYTES);
+      String metadataJson = null;
+      boolean shouldPersistMetadata =
+          switch (maxMetadataCacheBytes) {
+            case PolarisConfiguration.METADATA_CACHE_MAX_BYTES_INFINITE_CACHING -> true;
+            case PolarisConfiguration.METADATA_CACHE_MAX_BYTES_NO_CACHING -> false;
+            default -> {
+              metadataJson = TableMetadataParser.toJson(metadata);
```

Comment:
@snazy in this way, the re-serialization only happens when the feature is enabled